server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStatusApisIT.java

```diff
@@ -426,7 +427,16 @@ public void testGetSnapshotsWithSnapshotInProgress() throws Exception {
             .get();
         List<SnapshotInfo> snapshotInfoList = response1.getSnapshots();
         assertEquals(1, snapshotInfoList.size());
-        assertEquals(SnapshotState.IN_PROGRESS, snapshotInfoList.get(0).state());
+        SnapshotInfo snapshotInfo = snapshotInfoList.get(0);
+        assertEquals(SnapshotState.IN_PROGRESS, snapshotInfo.state());
+
+        SnapshotStatus snapshotStatus = client().admin().cluster().prepareSnapshotStatus().execute().actionGet().getSnapshots().get(0);
+        assertThat(snapshotInfo.totalShards(), equalTo(snapshotStatus.getIndices().get(indexName).getShardsStats().getTotalShards()));
+        assertThat(
+            snapshotInfo.successfulShards(),
+            lessThanOrEqualTo(snapshotStatus.getIndices().get(indexName).getShardsStats().getDoneShards())
```

Comment:
Same here: I had to relax the condition to `lessThanOrEqualTo` from `equalTo`, because the `getSnapshot`  and `snapshotStatus` API calls happen with a delay, so we can't guarantee that `successfulShards` and `doneShards` are actually the same